### PR TITLE
[SC-22] Upgrade to Aave v3.0.2

### PR DESCRIPTION
**Pull request by hexonaut**
This PR adds a deploy script for the Pool implementation and upgrades Aave to V3.0.2. This is the first step before running an upgrade spell. Upgrade PR: https://github.com/aave/aave-v3-core/pull/832/files

!!! You need to update Foundry to latest `foundryup` if you want to run the deploys as there is a library bug fix !!!

The review involves comparing to the Aave deployment that was activated in this proposal for mainnet: https://app.aave.com/governance/proposal/213/

All libraries and code should be confirmed to match.

To test the bytecode verification you need to set the `POOL_REVISION` to `0x2` inside `Pool` before compiling and running `validate-deploy.sh`.

### CODE REVIEW

**Comment by hexonaut** [comment]
Please note this is blocked on https://github.com/foundry-rs/foundry/issues/5330

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Did the following:
- Reviewed script
- Reviewed tests
- Reviewed AAVE submodule update diff for smart contracts, LGTM
- Ran a local sourcify and ran validation script successfully.

LGTM, just two small comments left and a question: Why do we have to change the REVISION value in Pool.sol to get the verification to work?

File: README.md [hunk 1]
```diff
@@ -9,6 +9,21 @@ Deploy Spark Lend: `ETH_RPC_URL=<YOUR RPC ENDPOINT> make deploy`
...
+
+### To Install
+
+1. Install local instance of sourcify via https://docs.sourcify.dev/docs/run-locally/#running-the-server
+2. Default `.env.dev` is mostly fine, but update `NODE_URL_MAINNET` to be a valid rpc endpoint.
+3. Start the server `npm run server:start`.
+
+### To Run
+
+Run `./validate-deploy.sh path/to/broadcast.json` (Be sure to `forge build` with proper settings first)
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply README.md:24]: Yeah this is fixed in the GC one as discussed. Reads from the file instead of `cast`.

**Review by lucas-manuel — APPROVED** [review]
Can merge once CI passes

**Pull request merged**